SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs

```diff
@@ -31,31 +35,63 @@ internal static string GetSecretForPlatform(string secrets, string platformIdent
             var parseErrorMessage = $"Error parsing key for '{platformIdentifier}'";
 
             var platformIndicator = platformIdentifier + "=";
+            var platformTargetIdicator = platformTargetIdentifier + "=";
             var secretIdx = secrets.IndexOf(platformIndicator, StringComparison.Ordinal);
-            if (secretIdx == -1)
+            var targetTokenIdx = secrets.IndexOf(platformTargetIdicator, StringComparison.Ordinal);
+            var targetIdx = secrets.IndexOf(targetKeyword.ToLower(), StringComparison.Ordinal);
+            if (secretIdx == -1 && targetTokenIdx == -1 && targetIdx == -1)
+            {
+                throw new AppCenterException(parseErrorMessage);
+            }
+            if (targetIdx >= 0 && secretIdx == -1 && targetTokenIdx == -1) 
```

Comment:
why not `!= -1`?

Reply:
Because we let that pass only if there's a string containing 'target='